Softeq.XToolkit.WhiteLabel.Droid/ViewComponents/ViewModelComponent.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using AndroidX.Fragment.App;
+
+namespace Softeq.XToolkit.WhiteLabel.Droid.ViewComponents
+{
+    /// <summary>
+    ///     Component, which encapsulates logic of storing/restoring ViewModel instance within FragmentManager
+    /// </summary>
+    /// <typeparam name="TViewModel">
+    ///     Type of the ViewModel instance
+    /// </typeparam>
+    public sealed class ViewModelComponent<TViewModel> where TViewModel : class
+    {
+        private TViewModel? _viewModel;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="ViewModelComponent{TViewModel}"/> class.
+        /// </summary>
+        /// <param name="viewModelKey">
+        ///     Key identifier, which will be used for storing/restoring TViewModel instance.
+        /// </param>
+        public ViewModelComponent(string viewModelKey)
+        {
+            ViewModelKey = viewModelKey;
+        }
+
+        /// <summary>
+        ///     Gets current instance of TViewModel, or throws exceptions if instance is not assigned.
+        /// </summary>
+        /// <exception cref="T:System.InvalidOperationException">
+        ///     ViewModel instance is not assigned either because it wasn't initialized using
+        ///     <see cref="ViewModelComponent{TViewModel}.Initialize"/> method, or wasn't restored properly using
+        ///     <see cref="ViewModelComponent{TViewModel}.RestoreViewModel"/> method.
+        /// </exception>
+        public TViewModel ViewModel => _viewModel ?? throw new InvalidOperationException("ViewModel instance is not assigned");
+
+        /// <summary>
+        ///     Gets key identifier witch used for storing/restoring TViewModel instance.
+        /// </summary>
+        public string ViewModelKey { get; }
+
+        /// <summary>
+        ///     Initializes <see cref="ViewModelComponent{TViewModel}"/> with an instance of ViewModel.
+        /// </summary>
+        /// <param name="viewModel">
+        ///     ViewModel instance to store.
+        /// </param>
+        public void Initialize(TViewModel viewModel)
+        {
```

Comment:
What about check when already initialized?

And case when:
- new
- .RestoreViewModel
- .Initialize

Reply:
> What about check when already initialized?
What is expected behaviour in this case in your opinion? I can't say for sure that it should be forbidden, but I can throw an exception in this case

Reply:
> And case when:
- new
- .RestoreViewModel
- .Initialize

This is not how this class is intended to be used. I'm not sure it can be handled within this class

Reply:
InvalidOperationException can be better than nothing in this case

Reply:
done